exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/SqlConverter.java

```diff
@@ -377,11 +377,15 @@ public RelRoot toRel(final SqlNode validatedNode) {
      * during creating new projects since it may cause changing data mode
      * which causes to assertion errors during type validation
      */
-    Hook.REL_BUILDER_SIMPLIFY.add(Hook.propertyJ(false));
+    Hook.Closeable closeable = Hook.REL_BUILDER_SIMPLIFY.add(Hook.propertyJ(false));
 
-    //To avoid unexpected column errors set a value of top to false
-    final RelRoot rel = sqlToRelConverter.convertQuery(validatedNode, false, false);
-    return rel.withRel(sqlToRelConverter.flattenTypes(rel.rel, true));
+    try {
+      //To avoid unexpected column errors set a value of top to false
+      final RelRoot rel = sqlToRelConverter.convertQuery(validatedNode, false, false);
+      return rel.withRel(sqlToRelConverter.flattenTypes(rel.rel, true));
+    } finally {
+      closeable.close();
```

Comment:
Could you please replace it with `try with resources` block?

Reply:
fixed and squashed